.eleventy.js

```diff
@@ -194,6 +194,102 @@ module.exports = function (config) {
     return filterTagList([...tagSet]);
   });
 
+  /**
+   * Create sidenavigation appropriate to folder structure
+   * An empty ".SIDENAV" file in a folder indicates the presence of a sidenav.
+   *  - 11ty will parse the folder. Example: content/docs/.SIDENAV
+   *  - This file can be empty. It just indicates the existence of navigation
+   * A ".SIDENAVCATEGORY" file describes information about the top level category.
+   *  - Example: content/docs/apps/.SIDENAVCATEGORY
+   *  - module.exports = {
+   *      name: "Compliance",
+   *      icon: "fa-check-square-o"
+   *    }
+   *   In your file front matter, to opt in to be included in a menu:
+   *   sidenav: true,
+   *   sidenavIdentifier: "docs",
+   */
+
+  config.addCollection("sideNavigation", function (collectionApi) {
+    const basePath = "content";
+    const mainFolders = fs
+      .readdirSync(basePath, {withFileTypes: true})
+      .filter((dirent) => {
```

Comment:
is `dirent` a typo of `direct` ?

Reply:
It's shorthand for "DIRectory ENTry" and probably comes from old C, but the JavaScript Type for it is also capitalized "Dirent".  Can change the variable to whatever though as it's used arbitrarily here.  I don't have much of an opinion on it.

Reply:
Interesting! I don't haver a classical C background so I didn't recognize it. Fine with me to keep!